### PR TITLE
feat: token gate decrypt

### PR DESCRIPTION
**Pull request by bigint**
🔒

### CODE REVIEW

**Comment by harish-sethuraman** [comment]
@bigint it failed

**Comment by bigint** [comment]
> @bigint it failed

failed to do `git add -A` 😭

File: apps/web/src/components/Publication/DecryptedPublicationBody.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  const canDecrypt = publication?.canDecrypt?.result;
+  const showMore = publication?.metadata?.content?.length > 450 && pathname !== '/posts/[id]';
+
+  const getDecryptedData = async () => {
+    if (!signer) {
+      return;
+    }
+
+    console.log(publication);
```
Comment (harish-sethuraman) [inline apps/web/src/components/Publication/DecryptedPublicationBody.tsx:34]:
```suggestion
```

File: apps/web/src/components/Publication/DecryptedPublicationBody.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    <div className="break-words">
+      <button
+        onClick={async (event) => {
+          event.stopPropagation();
+          await getDecryptedData();
+        }}
+      >
+        Unlock
+      </button>
+      {JSON.stringify(decryptedData)}
```
Comment (harish-sethuraman) [inline apps/web/src/components/Publication/DecryptedPublicationBody.tsx:55]: is this expected?

Comment (bigint) [reply apps/web/src/components/Publication/DecryptedPublicationBody.tsx:55]: Yes for now 😊

Comment (bigint) [reply apps/web/src/components/Publication/DecryptedPublicationBody.tsx:55]: Still not done, converting this PR to draft 🤣

File: apps/web/src/components/Publication/DecryptedPublicationBody.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          'whitespace-pre-wrap break-words leading-md linkify text-md'
+        )}
+      >
+        {publication?.metadata?.content}
+      </Markup>
+      {showMore && (
+        <div className="mt-4 text-sm text-gray-500 font-bold flex items-center space-x-1">
+          <EyeIcon className="h-4 w-4" />
+          <Link href={`/posts/${publication?.id}`}>Show more</Link>
+        </div>
```
Comment (harish-sethuraman) [inline apps/web/src/components/Publication/DecryptedPublicationBody.tsx:68]: Should we extract this showmore thing to another component? we do this for every long post and every long comment?

File: apps/web/src/components/Publication/SinglePublication.tsx [hunk 4]
```diff
@@ -76,7 +77,11 @@ const SinglePublication: FC<Props> = ({
           <HiddenPublication type={publication.__typename} />
         ) : (
           <>
-            <PublicationBody publication={rootPublication as LensterPublication} />
+            {publication?.metadata?.encryptionParams ? (
+              <DecryptedPublicationBody publication={rootPublication as LensterPublication} />
+            ) : (
+              <PublicationBody publication={rootPublication as LensterPublication} />
+            )}
```
Comment (harish-sethuraman) [inline apps/web/src/components/Publication/SinglePublication.tsx:84]: Why not infer them both at one place sir? 😅

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 5]
```diff
@@ -328,10 +329,14 @@ const NewPublication: FC<Props> = ({ publication }) => {
...
-    const tokenGatedSdk = await LensGatedSDK.create({ provider, signer, env: LensEnvironment.Mumbai });
+    const tokenGatedSdk = await LensGatedSDK.create({
+      provider,
+      signer,
+      env: LIT_PROTOCOL_ENVIRONMENT as any
+    });
     await tokenGatedSdk.connect({
       address: currentProfile.ownedBy,
-      env: LensEnvironment.Mumbai
+      env: LIT_PROTOCOL_ENVIRONMENT as any
```
[the remark this answers is not in the document]
Comment (bigint) [reply apps/web/src/components/Composer/NewPublication.tsx:339]: enum 🙈

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 6]
```diff
@@ -331,15 +339,39 @@ const NewPublication: FC<Props> = ({ publication }) => {
...
+    };
+
+    console.log(andCondition);
+
+    // Create the access condition
+    const accessCondition: AccessConditionOutput = {
+      and: andCondition
+    };
+
+    console.log(accessCondition);
```
Comment (harish-sethuraman) [inline apps/web/src/components/Composer/NewPublication.tsx:374]:
```suggestion
```

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 7]
```diff
@@ -331,15 +339,39 @@ const NewPublication: FC<Props> = ({ publication }) => {
...
+
+    // Combine the conditions
+    const andCondition: AndCondition = {
+      criteria: [
+        { collect: collectToView ? collectAccessCondition : null },
+        { follow: followToView ? followAccessCondition : null }
+      ]
+    };
+
+    console.log(andCondition);
```
Comment (harish-sethuraman) [inline apps/web/src/components/Composer/NewPublication.tsx:367]:
```suggestion
```

File: apps/web/src/components/Publication/DecryptedPublicationBody.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+  useEffect(() => {
+    if (canDecrypt) {
+      getDecryptedData();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [canDecrypt]);
+
+  if (!canDecrypt) {
+    return (
+      <Card className="text-sm rounded-xl w-fit p-8 shadow-sm" onClick={(event) => event.stopPropagation()}>
```
Comment (harish-sethuraman) [inline apps/web/src/components/Publication/DecryptedPublicationBody.tsx:58]:
```suggestion
      <Card className="text-sm rounded-xl w-fit p-8 shadow-sm">
```

Comment (bigint) [reply apps/web/src/components/Publication/DecryptedPublicationBody.tsx]: We need this, when user clicks on this card they should be taken to the post page

File: apps/web/src/components/Publication/SinglePublication.tsx (hunk 4 above)
Comment (bigint) [reply apps/web/src/components/Publication/SinglePublication.tsx]: Fixed here 🙇🏼 

https://github.com/lensterxyz/lenster/pull/1393/commits/5062908978f7b2a91953523a8544067b3e8d02f6

File: apps/web/src/components/Publication/DecryptedPublicationBody.tsx (hunk 3 above)
Comment (bigint) [reply apps/web/src/components/Publication/DecryptedPublicationBody.tsx]: We removed this for now for gated posts 🙇🏼

**Comment by bigint** [comment]
Going to merge this behind staffship feature flag 🙇🏼

**Pull request merged**